packages/AI/Core/src/generic/summarize.types.ts

```diff
@@ -11,8 +11,13 @@ export class SummarizeParams extends ChatParams {
 export class SummarizeResult extends BaseResult {
     text: string
     summaryText: string
-    constructor (text: string, summaryText: string, success: boolean, startTime: Date, endTime: Date) {
+    constructor (text: ChatMessageContent, summaryText: string, success: boolean, startTime: Date, endTime: Date) {
         super(success, startTime, endTime);
+
+        // if text is an array of ChatMessageContentBlock, filter it down to only text blocks and then concatenate them
+        if (text instanceof Array) {
+            text = text.filter((block: ChatMessageContentBlock) => block.type === 'text').map((block: ChatMessageContentBlock) => block.content).join('\n\n');
```

Comment:
The array element types are inferred from the array (`text`) since it's already been type-narrowed

```suggestion
            text = text.filter((block) => block.type === 'text').map((block) => block.content).join('\n\n');
```